.github/workflows/snap.yaml

```diff
@@ -114,7 +114,7 @@ jobs:
           bash test-script.sh
 
   publish:
-    if: inputs.publish == true && (github.ref == 'refs/heads/${{ inputs.branch-name }}' || startsWith(github.ref, 'refs/tags/'))
+    if: inputs.publish == true
```

Comment:
Any branch would be pushed to edge, not sure this is desirable.

Reply:
This is up to the workflow user to decide that.
From the perspective of this workflow we cannot know if we are on a PR or pushing on the main branch etc.
The user can write custom condition in the inputs.publish.